enhancements/security/openshift-image-policy.md

```diff
@@ -0,0 +1,338 @@
+    release.openshift.io/feature-set: TechPreviewNoUpgrade
+spec:
+  scopes:
+  - quay.io/openshift-release-dev/ocp-release
+  policy:
+    rootoftrust:
+      policyType: PublicKey
+      keydata: LS0tLS1CRUdJTiBQVUJMSUMgS0VZLS0tLS0KTUlJQ0lqQU5CZ2txaGtpRzl3MEJBUUVGQUFPQ0FnOEFNSUlDQ2dLQ0FnRUEzQzJlVGdJQUo3aGxveDdDSCtIcE1qdDEvbW5lYXcyejlHdE9NUmlSaEgya09ZalRadGVLSEtnWUJHcGViajRBcUpWYnVRaWJYZTZKYVFHQUFER0VOZXozTldsVXpCby9FUUEwaXJDRnN6dlhVbTE2cWFZMG8zOUZpbWpsVVovaG1VNVljSHhxMzR2OTh4bGtRbUVxekowR0VJMzNtWTFMbWFEM3ZhYmd3WWcwb3lzSTk1Z1V1Tk81TmdZUHA4WDREaFNoSmtyVEl5dDJLTEhYWW5BMExzOEJlbG9PWVJlTnJhZmxKRHNzaE5VRFh4MDJhQVZSd2RjMXhJUDArRTlZaTY1ZE4zKzlReVhEOUZ6K3MrTDNjZzh3bDdZd3ZZb1Z2NDhndklmTHlJbjJUaHY2Uzk2R0V6bXBoazRjWDBIeitnUkdocWpyajU4U2hSZzlteitrcnVhR0VuVGcyS3BWR0gzd3I4Z09UdUFZMmtqMnY1YWhnZWt4V1pFN05vazNiNTBKNEpnYXlpSnVSL2R0cmFQMWVMMjlFMG52akdsMXptUXlGNlZnNGdIVXYwaktrcnJ2QUQ4c1dNY2NBS00zbXNXU01uRVpOTnljTTRITlNobGNReG5xU1lFSXR6MGZjajdYamtKbnAxME51Z2lVWlNLeVNXOHc0R3hTaFNraGRGbzByRDlkVElRZkJoeS91ZHRQWUkrK2VoK243QTV2UVV4Wk5BTmZqOUhRbC81Z3lFbFV6TTJOekJ2RHpHellSNVdVZEVEaDlJQ1I4ZlFpMVIxNUtZU0h2Tlc3RW5ucDdZT2d5dmtoSkdwRU5PQkF3c1pLMUhhMkJZYXZMMk05NDJzSkhxOUQ1eEsrZyszQU81eXp6V2NqaUFDMWU4RURPcUVpY01Ud05LOENBd0VBQVE9PQotLS0tLUVORCBQVUJMSUMgS0VZLS0tLS0K
+```
+
+#### Installer
+
+As discussed in [the *Proposal* section](#proposal), the installer will grow a new, feature-gated `internalTestingImagePolicy` property driving ClusterVersion `overrides` extension.
+When set to `Disable`, the following `overrides` entry will be added to the ClusterVersion manifest:
+
+```yaml
+kind: ClusterImagePolicy
+group: config.openshift.io
+name: openshift
+unmanaged: true
+```
+
+##### Installer callers
+
+Hive, the assisted installer, and other actors who create install-configs will need to learn to set `internalTestingImagePolicy: Disabled` if they wish to test unsigned releases that enable the feature-gate.
+
+#### Implementation order
+
+To reduce risk, the recommended implementation order is:
+
+1. Machine-config operator [lifts the release payload scope restriction](#api-extensions).
+2. [Parallel Cincinnati work][OTA-1267] allows ART to push Sigstore signatures to `quay.io/openshift-release-dev/ocp-release:sha256-....sig`.
+3. Testing [the planned ClusterImagePolicy](#machine-config-operator) confirms that the approach works, possibly including [restricted-network use](#restricted-network-use).
+    If testing turns up gaps, round to see if we want to pivot or abandon this effort.
+4. [Installer `internalTestingImagePolicy` property](#installer).
+5. Tooling interacting with unsigned `quay.io/openshift-release-dev/ocp-release` release images ([CI, QE, assisted-installer, Hive, others?](#installer-callers)) updated to set `internalTestingImagePolicy`.
+6. Machine-config operator [adds the `openshift` ClusterImagePolicy manifest](#machine-config-operator).
+
+### Risks and Mitigations
+
+#### Changing our minds
```

Comment:
But that's not this enhancement's problem.  I just need to show that exiting from an OpenShift Sigstore policy is easy.  It would be the move-to-new-thing enhancement proposal that would have to explain how to move the existing clusters to the new thing, while fading out the old ways.

Reply:
I think that ignores the primary motivation for this enhancement - Red Hat's direction and timeline for eliminating simple signing. It is effecting design choices / limitations and we should appreciate the risk of functionally losing the ability to use simple signing in the near future.

Reply:
I've pushed 517e65ba -> 49e25242, adding some lines to the *Motivation* section expanding on why we like Sigstore more than OpenPGP as our current long-term strategy.

```markdown
Sigstore is both cheaper to run (because it's less custom than the existing OpenPGP / GPG system) and more powerful (because it verifies signatures on every pod launch, not just during OpenShift updates), and both of those motivate our desire to transition.
However, [the signing systems are orthogonal](#overlap-with-openpgp-signing), so technically we can run both in parallel or [change our minds](#changing-our-minds) and decide to refocus on OpenPGP or some new third signing method, as our understanding evolves.
This proposal covers the early stages of bringing in the Sigstore policy, and later work will cover things like removing OpenPGP to complete [a transitional overlap phase](#overlap-with-openpgp-signing).
```

I'm marking this thread resolved, but re-open or start a new one if I'm still missing something discussed in this thread.